src/test/java/org/springframework/data/elasticsearch/core/DefaultResultMapperTests.java

```diff
@@ -103,6 +106,42 @@ public void shouldMapGetRequestToObject() {
 		assertThat(result.getModel(), is("Grat"));
 		assertThat(result.getName(), is("Ford"));
 	}
+	
+	@Test
+	public void shouldMapImmutableCar() {
+		//Given
+		GetResponse response = mock(GetResponse.class);
+		final String id = "carId";
+		when(response.getId()).thenReturn(id);
+		when(response.getSourceAsString()).thenReturn(createJsonCar("Ford", "Grat"));
+
+		//When
+		ImmutableCar result = resultMapper.mapResult(response, ImmutableCar.class);
+
+		//Then
+		assertThat(result, notNullValue());
+		assertThat(result.getId(), is(id));
+		assertThat(result.getModel(), is("Grat"));
```

Comment:
Use constants (local var) for `Ford` and `Grat` as you do for `id`.
I see they are used in other test methods too - so they could be even proper constants in the test class.